src/mcmc/mcmc_functions.jl

```diff
@@ -0,0 +1,333 @@
+    else
+        write_galaxy_unit_flux(img0, pos, gal_frac_dev,
+                               gal_ab, gal_angle, gal_scale, src_pixels)
+    end
+
+    # create log like handle and return
+    function lnflux_loglike(lnflux)
+
+        # compute model image
+        rates = sky_pixels .+ exp(lnflux)*src_pixels
+
+        # sum per-pixel likelihood contribution
+        ll = 0.
+        H, W = size(rates)
+        for h in 1:H, w in 1:W
+            ll += poisson_lnpdf(img0.pixels[h,w], rates[h,w])
+        end
+
+        return ll
+    end
+
+    return lnflux_loglike
+end
+
+
+"""
+create vectorized star log flux log like function with multiple images
+"""
+function make_logflux_loglike(imgs::Array{Image},
+                              pos::Array{Float64, 1};
+                              is_star = true,
+                              gal_frac_dev::Float64 = .5,
+                              gal_ab::Float64       = .7,
+                              gal_angle::Float64    = .79,
+                              gal_scale::Float64    = 4.)
+
+    # create per image scalar log flux function (caches unit flux pixel image)
+    if is_star
+        scalar_funs = [make_single_image_logflux_loglike(img, pos;
+                          is_star=true)
+                       for img in imgs]
+    else
+        scalar_funs = [make_single_image_logflux_loglike(img, pos;
+                          is_star=false,
+                          gal_frac_dev=gal_frac_dev,
+                          gal_ab=gal_ab,
+                          gal_angle=gal_angle,
+                          gal_scale=gal_scale)
+                       for img in imgs]
+    end
+
+    function lnflux_loglike(lnflux::Vector{Float64})
+        # for each of those image/function pairs, compute flux-specific ll
+        ll = 0.
+        for i in 1:length(imgs)
+            b   = imgs[i].b
+            ll += scalar_funs[i](lnflux[b])
+        end
+        return ll
+    end
+
+    return lnflux_loglike
+end
+
+
+#############################
+# Priors and prior samplers #
+#############################
+
+const pp = Model.load_prior()
+
+"""
+prior over log fluxes (for stars and galaxies)
+"""
+function logflux_logprior(lnfluxes::Vector{Float64}; is_star::Bool=true)
+    # toggle star / galaxy
+    type_i = is_star ? 1 : 2
+
+    # convert to color space for prior calc
+    lnr, colors = logfluxes_to_colors(lnfluxes)
+
+    # compute brightness distribution
+    llr = logpdf(Normal(pp.r_μ[type_i], 2*pp.r_σ²[type_i]), lnr)
+
+    # compute color likelihoods (mixture model --- computes all component lls)
+    nc, nk, ns = size(pp.c_mean)
+    llk = [logpdf(MvNormal(pp.c_mean[:,k,type_i],
+                           pp.c_cov[:,:,k,type_i]), colors)
+           for k in 1:nk]
+    lnpik = log.(pp.k[:, type_i])
+    llc   = Model.logsumexp(llk .+ lnpik)
+
+    # add and return
+    return llr + llc
+end
+
+"""
+More numerically stable converter from log fluxes to colors
```

Comment:
Could we replace the unstable one with this one, to avoid redundant code? (Where is the unstable one?)